src/core/toga/widgets/optioncontainer.py

```diff
@@ -3,6 +3,100 @@
 from .base import Widget
 
 
+class OptionItem:
+    def __init__(self, widget):
```

Comment:
@freakboy3742 _index now is only updated by parent `OptionList` in `__getitem__` and `__iter__`. So `OptionItem` doesn't need to have control of its own index. Do you think it's valid?

ref to: https://github.com/beeware/toga/pull/832#discussion_r403426581

Reply:
Agreed - unless the item needs to know it's own position in the list, there's no reason to propegate `_index` to OptionItem.